### ISSUE TITLE
ERROR: Request failed: 500 column "key" does not exist

### ISSUE THREAD

**Issue by wsescu** [dalibo/temboard#888]
Dear temBoard support

whenever I click on "Rollbacked transactions" and select a different time range other than the default, e.g. Last 2 days, or Last 7 days, I do get the following error in the temboard.log

```
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR: Unhandled Error:
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR: Traceback (most recent call last):
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 227, in error_middleware
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:     return func(request, *args)
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 408, in user_instance_middleware
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:     return func(request, address, port, *args)
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 267, in instance_middleware
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:     return callable_(request, *args)
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 431, in user_middleware
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:     return func(request, *args)
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/plugins/monitoring/handlers/monitoring.py", line 164, in data_metric
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:     key=key,
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/plugins/monitoring/chartdata.py", line 872, in get_metric_data_csv
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:     data_buffer)
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR: ProgrammingError: column "key" does not exist
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR: LINE 6: ...021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'blo...
2021-01-21 08:04:37,806 temboardui[11275]: [web] ERROR:                                                              ^
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: Unhandled Error:
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: Traceback (most recent call last):
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 227, in error_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 408, in user_instance_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, address, port, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 267, in instance_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return callable_(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 431, in user_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/plugins/monitoring/handlers/monitoring.py", line 164, in data_metric
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     key=key,
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/plugins/monitoring/chartdata.py", line 872, in get_metric_data_csv
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     data_buffer)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: ProgrammingError: column "key" does not exist
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: LINE 6: ...021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'rep...
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:                                                              ^
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: Unhandled Error:
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: Traceback (most recent call last):
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 227, in error_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 408, in user_instance_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, address, port, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 267, in instance_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return callable_(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 431, in user_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/plugins/monitoring/handlers/monitoring.py", line 164, in data_metric
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     key=key,
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/plugins/monitoring/chartdata.py", line 872, in get_metric_data_csv
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     data_buffer)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: ProgrammingError: column "key" does not exist
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: LINE 6: ...021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'pos...
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:                                                              ^
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: Unhandled Error:
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: Traceback (most recent call last):
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 227, in error_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 408, in user_instance_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, address, port, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 267, in instance_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return callable_(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/web.py", line 431, in user_middleware
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     return func(request, *args)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/plugins/monitoring/handlers/monitoring.py", line 164, in data_metric
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     key=key,
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:   File "/usr/lib/python2.7/site-packages/temboardui/plugins/monitoring/chartdata.py", line 872, in get_metric_data_csv
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:     data_buffer)
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: ProgrammingError: column "key" does not exist
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR: LINE 6: ...021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'tem...
2021-01-21 08:04:37,813 temboardui[11275]: [web] ERROR:                                                              ^
2021-01-21 08:04:37,814 temboardui[11275]: [web] ERROR: Request failed: 500 column "key" does not exist
2021-01-21 08:04:37,814 temboardui[11275]: [web] ERROR: LINE 6: ...021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'blo...
2021-01-21 08:04:37,814 temboardui[11275]: [web] ERROR:                                                              ^
2021-01-21 08:04:37,814 temboardui[11275]: [web] ERROR: .
2021-01-21 08:04:37,815 temboardui[11275]: [web] ERROR: Request failed: 500 column "key" does not exist
2021-01-21 08:04:37,815 temboardui[11275]: [web] ERROR: LINE 6: ...021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'rep...
2021-01-21 08:04:37,815 temboardui[11275]: [web] ERROR:                                                              ^
2021-01-21 08:04:37,815 temboardui[11275]: [web] ERROR: .
2021-01-21 08:04:37,815 temboardui[11275]: [web] ERROR: Request failed: 500 column "key" does not exist
2021-01-21 08:04:37,815 temboardui[11275]: [web] ERROR: LINE 6: ...021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'pos...
2021-01-21 08:04:37,815 temboardui[11275]: [web] ERROR:                                                              ^
2021-01-21 08:04:37,815 temboardui[11275]: [web] ERROR: .
2021-01-21 08:04:37,816 temboardui[11275]: [web] ERROR: Request failed: 500 column "key" does not exist
2021-01-21 08:04:37,816 temboardui[11275]: [web] ERROR: LINE 6: ...021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'tem...
2021-01-21 08:04:37,816 temboardui[11275]: [web] ERROR:                                                              ^
2021-01-21 08:04:37,816 temboardui[11275]: [web] ERROR: .
```

And the following errors in the PostgreSQL alert.log

```
2021-01-21 08:04:37.812 CET [24672] ERROR:  column "key" does not exist at character 262
2021-01-21 08:04:37.812 CET [24672] STATEMENT:  COPY(
        SELECT
            datetime AS date,
            SUM((record).n_rollback) AS rollback
        FROM metric_xacts_30m_current
        WHERE instance_id = 21 AND datetime >= '2021-01-19T07:04:37.736000+00:00'::timestamptz AND datetime <= '2021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'repmgr'
        GROUP BY datetime, instance_id ORDER BY datetime
                ) TO STDOUT WITH CSV HEADER
2021-01-21 08:04:37.812 CET [29253] ERROR:  column "key" does not exist at character 262
2021-01-21 08:04:37.812 CET [29253] STATEMENT:  COPY(
        SELECT
            datetime AS date,
            SUM((record).n_rollback) AS rollback
        FROM metric_xacts_30m_current
        WHERE instance_id = 21 AND datetime >= '2021-01-19T07:04:37.736000+00:00'::timestamptz AND datetime <= '2021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'postgres'
        GROUP BY datetime, instance_id ORDER BY datetime
                ) TO STDOUT WITH CSV HEADER
2021-01-21 08:04:37.813 CET [18105] ERROR:  column "key" does not exist at character 262
2021-01-21 08:04:37.813 CET [18105] STATEMENT:  COPY(
        SELECT
            datetime AS date,
            SUM((record).n_rollback) AS rollback
        FROM metric_xacts_30m_current
        WHERE instance_id = 21 AND datetime >= '2021-01-19T07:04:37.736000+00:00'::timestamptz AND datetime <= '2021-01-21T07:04:37.736000+00:00'::timestamptz AND key = 'template1'
        GROUP BY datetime, instance_id ORDER BY datetime
                ) TO STDOUT WITH CSV HEADER
```

I am running temboard 7.5 and agent 7.5.

Cheers
William

**Comment by pgiraud** [comment]
Hi @wsescu and thanks for thre report.
I confirm that this is a bug and I can reproduce it.
I will try to fix it as soon as possible.

**Comment by wsescu** [comment]
Thanks @pgiraud for taking care of this one.

**Comment by pgiraud** [comment]
Fix shipped with 7.6.
